HIPPO-2: Hemi Economic Model (#7)
hemilabs/hippos: hippo-002.md

# HIPPO-002

```
HIPPO: 2
Title: Hemi Economic Model
Author: Jeff Garzik <jeff@hemi.xyz>
        Max Sanchez <max@hemi.xyz>
Comments-Summary: No comments yet.
Status: Active
Type: Network
Created: 2025-10-24
```

## Background and Motivation

> The Hemi blockchain connects Bitcoin and Ethereum - enabling new  

Reviewer: Was it intentional to add two spaces to the end of each line?
This prevents GitHub (and most other markdown renderers) from unwrapping the lines, and forces the following line onto a newline.

I think this makes the document harder to read when it is rendered, however it is not a super big issue.

Author: Let's get this merged and then I'll do another PR with some formatting tweaks

the protocol, and directing protocol asset flows within this economic model.

Users stake $HEMI for a period of 12 days to 4 years, and receive an NFT  
representing their staked position. Longer lockups receive higher weight  
in the system, representing the increased incentive alignment with the  
long-term success of Hemi.

The weight of a position is calculated based on its remaining lockup time;  
for example, a position with 4 years remaining has twice the weight of a   
position with 2 years remaining, etc. Users can extend the lockups on  
existing positions to increase their weight and incentives.

As Hemi evolves, owners of veHEMI positions will earn additional incentives for:
- Participating in governance votes (or delegating)
- Operating decentralized infrastructure:
    - Block sequencing
    - Data publication to Ethereum (DA & state roots)
    - hBitVM covenant emulation and vault operation
    - Providing shared sequencing and DA services to Chainbuilder chains
- Providing cross-chain liquidity
- Providing economic security to third-party protocols
- Receiving yield from the vote market for directing incentives

> When veHemi positions are used for economic security (decentralized  

Reviewer: veHemi -> veHEMI?

infrastructure, cross-chain liquidity, and securing third-party  
protocols), they are subject to being slashed for misbehavior.  
Depending on the specific security needs of each protocol, the  
underlying $HEMI in the slashed position may be burnt, sent to  
either the STP or POL, and/or distributed to other protocol  
participants.

## Phases of Hemi Economic Model
The Hemi Economic Model will be implemented across several different  
phases, each building upon the last to further decentralize control of  
the economic model, as well as introduce new mechanics that serve to  
incentivize more types of activity and create more incentive alignment  
between the participants in the Hemi ecosystem.

The phases of the economic model will be:

> - **Phase 1 (Today):** Protocol Fees as veHEMI Incentives & Burn

Reviewer: Would it make sense to replace "Today" with an actual date here?

 - Conversion of 2.77 ETH to 198,537.44 HEMI:
   - [https://explorer.hemi.xyz/tx/0xf8e13b27f9085f48d66bde0ba68827c0382e9bade37aabef29ed01bb5ca81e39](https://explorer.hemi.xyz/tx/0xf8e13b27f9085f48d66bde0ba68827c0382e9bade37aabef29ed01bb5ca81e39)
   - [https://explorer.hemi.xyz/tx/0xdb2d914bc31b8c52109180550a53115dcb752bb16a7d8c2b5acd9be9fa383861](https://explorer.hemi.xyz/tx/0xdb2d914bc31b8c52109180550a53115dcb752bb16a7d8c2b5acd9be9fa383861)
 - Burning of 98,216.75 HEMI:
   - [https://etherscan.io/tx/0x7aa573ea8dbf20ace5477d2631609cce53990aba7f5dfbe9b336b9c22a7c751c](https://etherscan.io/tx/0x7aa573ea8dbf20ace5477d2631609cce53990aba7f5dfbe9b336b9c22a7c751c)

Moving forward, excess protocol fees will be routinely collected and converted  
to $HEMI and hemiBTC to be distributed as veHEMI rewards, and $HEMI which is burnt.

Only the owner of a veHEMI NFT position can claim the rewards, and the rewards  
must be claimed per NFT; holders of multiple veHEMI positions will need to perform  
one claim for each position.

> UI components will be added to the Hemi Portal to allow users to clai mrewards on  

Reviewer 2: spelling mistake

```suggestion
UI components will be added to the Hemi Portal to allow users to claim rewards on  
```

Author: Fixed

their veHEMI positions, and a future update will show APY estimations.

## Compatibility
This upgrade only requires smart contract deployments and configuration; it  
is not a change to the Hemi protocol and does not require node operators to  
update their software.

## Changelog

2025-10-24 Initial version.

> 2025-10-30 Converted to full Hemi Economic Model details.

Reviewer: In the rendered markdown, these display on a single line instead of separate lines. I think this can be fixed by adding a blank line between or adding `<br/>` to the end of the first line.

```suggestion
2025-10-24 Initial version.

2025-10-30 Converted to full Hemi Economic Model details.
```
or
```suggestion
2025-10-24 Initial version.<br/>
2025-10-30 Converted to full Hemi Economic Model details.
```
or two spaces on the end of the first line:
```suggestion
2025-10-24 Initial version.  
2025-10-30 Converted to full Hemi Economic Model details.
```

Author: Fixed

# HIPPO-002

```  
HIPPO: 2  
Title: Initial veHEMI Staking Rewards  
Author: Max Sanchez <max@hemi.xyz>  
Comments-Summary: No comments yet.  
Status: New  
Type: Network  
Created: 2025-10-24  
```  

## Background and Motivation

veHEMI is the native staking system for $HEMI on the Hemi network, with  
staking periods between 12 days and 4 years supported in 6-day epochs.

Depositing $HEMI into veHEMI mints an NFT which represents the staked  

> position. Each NFT has a weight based on the total tokens staked and  

Reviewer 3: I'm a little confused reading this part.  When you deposit $HEMI, you choose an amount to stake and a total lockup time and then recieve an NFT with a weight based on the amount and your lockup time, correct?  Saying `remaining lockup time` feels like a slightly backwards way of thinking about it,, doesn't it?  Or does the weight change such that at `t1` your NFT has weight `X` and then at `t1+lockup/2` you have wieght `X/2`?  Either way, I think it could be phrased a little clearer.

Author: The weight changes over time; for example if you locked up 1 $HEMI for 4 years, your position initially has a weight of 1, then after 2 years it has a weight of 0.5, etc. until you get to the unlock where it has a weight of 0.

> the time remaining in the lockup. This weight increases linearly with  the

Reviewer 3: extra space before `the`

amount of remaining time, such that a lockup with 1 year remaining  
has twice the weight of a lockup with 6 months remaining.

In the future, veHEMI will serve as the economic security system behind  
many of the protocol's features including governance, decentralized

> sequencing,  data publication to Ethereum, hBitVM covenant emulation,

Reviewer 2: There's an extra spacing here

```suggestion
sequencing, data publication to Ethereum, hBitVM covenant emulation,
```

> cross-chain  liquidity systems, and more.  In order to properly incentivize

Reviewer 2: Extra space here

```suggestion
cross-chain liquidity systems, and more.  In order to properly incentivize
```

> staking in preparatiion for these features, this HIPPO proposes the initial

Reviewer 3: preparation has too many i's in it

> rewards to be distributed  to veHEMI holders.

Reviewer 2: another extra space

```suggestion
rewards to be distributed to veHEMI holders.
```

## Specification
The initial quantity of $HEMI to be distributed to veHEMI stakers for  
the period of August 30th, 2025 to October 24th, 2025 will be 500,000  
$HEMI.

The tokens will be claimable based on a weighted average of each NFT's  
weight on a daily snapshot.

These snapshots will start at unix epoch 1756512000 (August 30th, 2025  
at 12:00am GMT) and run through 1761264000 (October 24th, 2025 at   

> 12:00am GMT) inclusive, which is a total of 56 days.

Reviewer 3: Since epochs are 6 days long, is it okay that this period is not divisible by 6?

Author: Yep, rewards can be calculated on a different schedule than the actual epochs.

The total 500,000 $HEMI will be split equally across each of the 56  
snapshots; equaling approximately 8928.5714 $HEMI per day.

This will be implemented by a reward contract with a claim function which
calculates the relative weight of a claimant's veHEMI position compared to
the total weight of all positions at each snapshot, and distributes the
proportional amount of rewards to the owner.

Only the owner of a veHEMI position can claim the rewards, and rewards
must be claimed per NFT; holders of multiple veHEMI positions in the same
address will need to perform one claim for each veHEMI position.

All active veHEMI positions are treated equally for this reward distribution.

UI components will be added to the Hemi Portal to allow users to claim rewards
on their veHEMI positions, as well as show estimated APY.

> ## Compatability

Reviewer 2: ```suggestion
## Compatibility
```

spelling mistake
